### PR TITLE
lib: Emit OSC 7 escape sequence (current host and working directory) for many Unix terminal emulators

### PR DESCRIPTION
**Pull request by ntninja**
## Standards checklist:

- [x] The PR title is descriptive.
- [x] The PR doesn't replicate another PR which is already open.
- [x] I have read the contribution guide and followed all the instructions.
- [ ] The code follows the code style guide detailed in the wiki.
   * Don't know, but the code follows the style already present in the respective file.
- [x] The code is mine or it's from somewhere with an MIT-compatible license.
- [x] The code is efficient, to the best of my ability, and does not waste computer resources.
- [x] The code is stable and I have tested it myself, to the best of my abilities.

## Changes:

- Report current host and working directory to many Unix terminal emulators in addition to iTerm/iTerm2

## Other comments:

Previously this was only emitted on on macOS with Apple's iTerm.app (and
compatible clones like iTerm2), but it is used by other terminal emulators
as well to obtain the actual current working directory wiht symlinks intact.
All non-supporting terminal emulators tested gracefully ignored this value,
so emit this on these as well in case they (or some other app masquarading
as them) add future support for this value.

### CODE REVIEW

File: lib/termsupport.zsh [hunk 1]
```diff
@@ -110,28 +110,52 @@ add-zsh-hook precmd omz_termsupport_precmd
...
+  # Percent-encode the host and path names.
+  local URL_HOST URL_PATH
+  URL_HOST="$(omz_urlencode -P $HOST)" || return 1
+  URL_PATH="$(omz_urlencode -P $PWD)" || return 1
+
+  # common control sequence (OSC 7) to set current host and path
+  printf "\e]7;%s\a" "file://${URL_HOST}${URL_PATH}"
+}
+
+#FIXME: Use a precmd hook instead of a chpwd hook to avoid contaminating output – why!?
```
Comment (ntninja) [inline lib/termsupport.zsh:158]: This one I'm unsure about:

The previous code had `precmd` with the above comment, but I sincerely do not understand what they mean by *avoid contaminating out* here?

**Comment by ntninja** [comment]
Also is the commit supposed to be `feat` or `fix` in this case?

**Comment by ngkz** [comment]
foot also supports OSC 7. Can you add that to the `${TERM}` case?

File: lib/termsupport.zsh [hunk 2]
```diff
@@ -110,28 +110,52 @@ add-zsh-hook precmd omz_termsupport_precmd
...
+#
+# Identifies the directory using a file: URI scheme, including
+# the host name to disambiguate local vs. remote paths.
+function omz_termsupport_chpwd {
+  emulate -L zsh
+
+  [[ "$INSIDE_EMACS" == *term* ]] && return
+
+  case "${TERM}" in
+    xterm*|putty*|rxvt*|konsole*|mlterm*|alacritty|screen*|tmux*)
```
Comment (christianparpart) [inline lib/termsupport.zsh:138]:
```suggestion
    xterm*|putty*|rxvt*|konsole*|mlterm*|alacritty|screen*|tmux*|contour*)
```
While being at it (in #11599), you could extend it here accordingly as well. :)

File: lib/termsupport.zsh [hunk 3]
```diff
@@ -110,28 +110,52 @@ add-zsh-hook precmd omz_termsupport_precmd
...
-  add-zsh-hook precmd update_terminalapp_cwd
-  # Run once to get initial cwd set
-  update_terminalapp_cwd
-fi
+# Emits the control sequence to notify many terminal emulators
+# of the cwd
+#
+# Identifies the directory using a file: URI scheme, including
+# the host name to disambiguate local vs. remote paths.
+function omz_termsupport_chpwd {
```
Comment (christianparpart) [inline lib/termsupport.zsh:132]: This name, `omz_termsupport_chpwd`, is badly named. `pwd` stands for `print working dir`. what would probably be a better fit is `omz_termsupport_set_cwd` (with `cwd` as in `current working directory`). That would be much more common sense.

Comment (christianparpart) [reply lib/termsupport.zsh:132]: ok, but then the hook is named very badly. I don't want to force you to switch it though. Thx for the heads-up ;)

File: lib/termsupport.zsh (hunk 2 above)
Comment (christianparpart) [reply lib/termsupport.zsh:138]: Why was this comment ignored?

Comment (christianparpart) [reply lib/termsupport.zsh:138]: Never mind. I can't read. You redid the commit elsewhere. Thanks :)

**Pull request closed without merging**